tests/ami/test_data.py

```diff
@@ -0,0 +1,386 @@
+            "population": "-",
+            "start_date": "2026-12-18T23:00:00+00:00",
+            "end_date": "2027-01-03T23:00:00+00:00",
+            "location": "Lille",
+            "zones": "Zone B",
+            "annee_scolaire": "2026-2027",
+        },
+        {
+            "description": "Vacances de Printemps",
+            "population": "-",
+            "start_date": "2027-04-16T22:00:00+00:00",
+            "end_date": "2027-05-02T22:00:00+00:00",
+            "location": "Reims",
+            "zones": "Zone B",
+            "annee_scolaire": "2026-2027",
+        },
+    ]
+    fake_holidays_data1 = {"total_counts": 2, "results": fake_holidays_data_2526}
+    fake_holidays_data2 = {
+        "total_counts": 4,
+        "results": fake_holidays_data_2526 + fake_holidays_data_2627,
+    }
+
+    holidays_2526 = [
+        {
+            "description": "Vacances de Noël",
+            "start_date": "2025-12-19T23:00:00Z",
+            "end_date": "2026-01-04T23:00:00Z",
+            "zones": "Zone B",
+            "emoji": "🎄",
+        },
+        {
+            "description": "Vacances d'Hiver",
+            "start_date": "2026-02-13T23:00:00Z",
+            "end_date": "2026-03-01T23:00:00Z",
+            "zones": "Zone B",
+            "emoji": "❄️",
+        },
+    ]
+    holidays_2627 = [
+        {
+            "description": "Vacances de Noël",
+            "start_date": "2026-12-18T23:00:00Z",
+            "end_date": "2027-01-03T23:00:00Z",
+            "zones": "Zone B",
+            "emoji": "🎄",
+        },
+        {
+            "description": "Vacances de Printemps",
+            "start_date": "2027-04-16T22:00:00Z",
+            "end_date": "2027-05-02T22:00:00Z",
+            "zones": "Zone B",
+            "emoji": "🌸",
+        },
+    ]
+    fake_holidays_data1 = {
+        "total_counts": len(fake_holidays_data_2526),
+        "results": fake_holidays_data_2526,
+    }
+    fake_holidays_data2 = {
+        "total_counts": len(fake_holidays_data_2526 + fake_holidays_data_2627),
+        "results": fake_holidays_data_2526 + fake_holidays_data_2627,
+    }
+
+    # get holidays of current school year only
+    httpx_mock.add_response(
+        url=httpx.URL(
+            "https://data.education.gouv.fr/api/explore/v2.1/catalog/datasets/fr-en-calendrier-scolaire/records",
+            params={
+                "where": (
+                    "end_date >= date'2025-09-01' AND start_date < date'2026-09-15' AND "
+                    "(zones = 'Zone A' AND location = 'Bordeaux' OR zones = 'Zone B' "
+                    "AND location = 'Lille' OR zones = 'Zone C' AND location = 'Versailles') "
+                    "AND population IN ('-', 'Élèves')"
+                ),
+                "order_by": "start_date",
+                "limit": 100,
+            },
+        ),
+        json=fake_holidays_data1,
+    )
+    response = test_client.get("/data/holidays", params={"start_date": "2025-09-01"})
```

Comment:
Nit: if what you're trying to test here is the span of the query (start and end date) given the given "start date", then maybe you don't need to go through all that?
Maybe just mocking the queries and making sure they were called with the given params?
You wouldn't even need to double check the result of the call to our holidays api, as you've tested that in the previous test?